pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -279,6 +299,48 @@ private RebalancePreCheckerResult checkDiskUtilization(Map<String, Map<String, S
         : RebalancePreCheckerResult.error(message.toString());
   }
 
+  private RebalancePreCheckerResult checkRebalanceConfig(RebalanceConfig rebalanceConfig, TableConfig tableConfig,
+      Map<String, Map<String, String>> currentAssignment, Map<String, Map<String, String>> targetAssignment) {
+    StringBuilder message = new StringBuilder();
+    boolean pass = true;
+    if (rebalanceConfig.isBestEfforts()) {
+      pass = false;
+      message.append("bestEfforts is enabled, only enable it if you know what you are doing\n");
+    }
+    List<String> segmentsToMove = SegmentAssignmentUtils.getSegmentsToMove(currentAssignment, targetAssignment);
+
+    int numReplicas = Integer.MAX_VALUE;
+    for (String segment : segmentsToMove) {
+      numReplicas = Math.min(targetAssignment.get(segment).size(), numReplicas);
+    }
```

Comment:
Can you move all of this into:

```
if (rebalanceConfig.isDowntime()) {

}
```

That way we don't need to calculate this at all if `downtime=false`